src/SDK/Trace/SpanProcessor/BatchSpanProcessorBuilder.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Trace\SpanProcessor;
+
+use OpenTelemetry\SDK\Common\Time\ClockFactory;
+use OpenTelemetry\SDK\Common\Time\ClockInterface;
+use OpenTelemetry\SDK\Metrics\MeterProviderInterface;
+use OpenTelemetry\SDK\Trace\SpanExporterInterface;
+
+class BatchSpanProcessorBuilder
+{
+    private SpanExporterInterface $exporter;
+    private ?MeterProviderInterface $meterProvider = null;
+    private ?ClockInterface $clock = null;
+
+    public function __construct(SpanExporterInterface $exporter)
+    {
+        $this->exporter = $exporter;
+    }
+
+    public function setMeterProvider(MeterProviderInterface $meterProvider): self
+    {
+        $this->meterProvider = $meterProvider;
+
+        return $this;
+    }
+
+    public function setClock(ClockInterface $clock): self
```

Comment:
I don't think we should make the clock configurable in the builder as this is specific to the current implementation. If somebody needs this flexibility, they can use the constructor directly.